libbpf-rs/src/streams.rs

```diff
@@ -0,0 +1,105 @@
+use core::ffi::c_void;
+use std::io;
+
+use crate::libbpf_sys::bpf_prog_stream_read_opts;
+
+use crate::Error;
+use crate::Result;
+
+/// [`Stream`] instances.
+///
+/// `streams`s are BPF descriptors that provide a character output
+/// interface from the BPF program to userspace.
+#[derive(Debug)]
+pub struct Stream {
+    fd: i32,
+    stream_id: u32,
+    bytes_per_read: usize,
+}
+
+/// Default BPF stdout stream id.
+pub const BPF_STDOUT: u32 = 1;
+/// Default BPF stderr stream id.
+pub const BPF_STDERR: u32 = 2;
+
+/// Config options, currently empty and here for forward compatibility.
+struct StreamReadOpts {
+    sz: usize,
+}
```

Comment:
I kind of fail to see the purpose of this type. It's not public, so users can't use it. So we don't need forward compat. And if it's empty, then we don't need it at all?